_source/_posts/2022-02-28-shopping-app-with-node-minio.md

```diff
@@ -0,0 +1,709 @@
+      <div class="caption">
+        <h3>Product Name</h3>
+        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
+        <p>
+          <a href="#" class="btn btn-primary">Buy Now!</a> <a href="#" class="btn btn-default">More Info</a>
+        </p>
+      </div>
+    </div>
+  </div>
+{{/each}}
+</div>
+```
+{% endraw %}
+
+{% img
+blog/shopping-app-with-node-minio/home.png
+alt:"Home page"
+width:"800" %}{: .center-image }
+
+The app needs several entry points such as login, register, forgot-password and unlock. We will specify the configuration of the sign-in widget in each of these handlebars with the appropriate `flow:<entry point>` option. The `flow` config option is new in the sign-in widget and it's currently in beta.
+
+
+### Login page
+
+Create a file called `Login.handlebars`and add the following code to it:
+
+{% raw %}
+```html
+{{#if userContext}}
+<h1 class="text-center">Hi {{userContext.userinfo.given_name}}!</h1>
+<div class="d-flex justify-content-center"></div>
+
+<!-- Title -->
+<div class="row">
+  <div class="col-lg-12">
+    <h3>Latest Products</h3>
+  </div>
+</div>
+<!-- /.row -->
+
+<!-- Page Features -->
+<div class="row text-center">
+  {{#each url}}
+    <div class="col-md-3 col-sm-6 hero-feature">
+      <div class="thumbnail">
+        <img src="{{this}}" max-height=200 max-width=200 alt="">
+        <div class="caption">
+          <h3>Product Name</h3>
+          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
+          <p>
+            <a href="#" class="btn btn-primary">Buy Now!</a> <a href="#" class="btn btn-default">More Info</a>
+          </p>
+        </div>
+      </div>
+    </div>
+  {{/each}}
+
+</div>
+<!-- /.row -->
+{{else}}
+
+<h1 class="text-center">Please Log In</h1>
+<div id="okta-login-container">
+</div>
+<script src="../js/okta-config.js"></script>
+<script>
+  signInWidgetConfig = {
+    language: 'en',
+    i18n: {
+      'en': {
+        'primaryauth.title': 'Sign In',
+        'primaryauth.submit': 'Sign In',
+      }
+    },
+    signOutLink: window.location.origin + '/logout',
+    features: {
+      registration: true,
+      rememberMe: true,
+      router: true,
+    },
+    baseUrl: window.location.origin,
+    flow: 'login',
+    el: '#okta-login-container',
+    clientId: credentials.clientId,
+    redirectUri: window.location.origin + '/callback',
+    authParams: {
+      issuer: credentials.domain + '/oauth2/default',
+      responseType: ['code'],
+      useInteractionCodeFlow: true,
+      scopes: ['openid', 'email', 'profile'],
+    },
+  };
+  var oSignIn = new OktaSignIn(signInWidgetConfig);
+  oSignIn.showSignInAndRedirect({
+    el: '#okta-login-container'
+  }).catch(function(error) {
+    alert('Catch unknown errors here which widget does not support');
+  });
+</script>
+{{/if}}
+```
+{% endraw %}
+
+The flow parameter is not needed for the login flow since its the default flow for the sign in widget.
```

Comment:
```suggestion
The flow parameter is not needed for the login flow since it's the default flow for the Sign-In Widget.
```
apostrophe/typo & capitalization